### PR TITLE
Add support for all stacks

### PR DESCRIPTION
**Pull request by jpena-r7**
## Summary

This change:
* Adds support any stack
* Uses upstream dependencies from golang for all versions
* Adds a non-paketo builder to the integration tests verify `*` stack support
* Updates the regex for one of the tests that was failing to handler longer durations.

## Use Cases

This allows the go-dist buildpack to be used on any stack.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by robdimsdale — DISMISSED** [review]
Not a Go maintainer, but this looks good to me.

Out of curiosity, did you see the test take longer (and hence fail, before your regex change)? I wouldn't have expected the tests to take substantially longer against the non-Paketo stack because the buildpack is not doing anything different than it does on the Paketo stacks, right? Not a problem or a blocker, just curious.

**Comment by jpena-r7** [comment]
That is a good question. I didn't expect it to require any other changes, but it failed because of this: `Completed in 1m13.783s`.

It didn't take a substantially long time to finish, but my machine might be a bit sluggish because I have too many things running on it. After looking at the other regexes I can see that they are written to account for the letters that denote `m` for minutes.

File: integration.json [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {
-  "builders": ["paketobuildpacks/builder:buildpackless-base", "paketobuildpacks/builder-jammy-buildpackless-base"],
+  "builders": [
+    "paketobuildpacks/builder:buildpackless-base",
+    "paketobuildpacks/builder-jammy-buildpackless-base",
+    "index.docker.io/jericop/amazonlinux-builder:base"
```
[the remark this answers is not in the document]
Comment (jpena-r7) [reply integration.json:5]: Sounds good to me. I added it to show that I tested it on a non-paketo stack to verify all tests still pass. Please confirm and I can remove it.

[the remark this answers is not in the document]
Comment (ForestEckhardt) [reply integration.json:5]: Tests are green in the workflow which is good enough for me!

**Comment by jpena-r7** [comment]
I just pushed another change with the requested updates. I also updated `buildpack.toml` after using jam to update the dependencies. I had originally done it manually, but that was much easier and less error prone. Thanks @fg-j for the help with that.

File: integration.json (hunk 1 above)
Comment (ForestEckhardt) [inline integration.json:5]:
```suggestion
```

File: integration.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
-  "builders": ["paketobuildpacks/builder:buildpackless-base", "paketobuildpacks/builder-jammy-buildpackless-base"],
+  "builders": [
+    "paketobuildpacks/builder:buildpackless-base",
+    "paketobuildpacks/builder-jammy-buildpackless-base",
```
Comment (ForestEckhardt) [inline integration.json:4]:
```suggestion
    "paketobuildpacks/builder-jammy-buildpackless-base"
```
It always gets me!!!!!

Comment (jpena-r7) [reply integration.json]: No worries. It gets me too. 😄

**Review by ForestEckhardt — APPROVED** [review]

**Review by fg-j — APPROVED** [review]

**Pull request merged**